pyfr/util.py

```diff
@@ -168,3 +169,39 @@ def match_paired_paren(delim, n=5):
     rgt = rf'\{close}{ocset}*?)*?'
 
     return lft*n + mid + rgt*n
+
+
+def file_path_gen(basedir, basename):
+    ns = 0 
+
+    # See if the basename appears to depend on {n} 
+    if re.search('{n[^}]*}', basename): 
+        # Quote and substitute 
+        bn = re.escape(basename) 
+        bn = re.sub(r'\\{n[^}]*\\}', r'(\\s*\\d+\\s*)', bn) 
+        bn = re.sub(r'\\{t[^}]*\\}', r'(?:.*?)', bn) + '$' 
+        for f in os.listdir(basedir): 
+            if (m := re.match(bn, f)): 
+                ns = max(ns, int(m[1]) + 1)
+
+    t = yield
+
+    for n in it.count(ns): 
+        t = yield os.path.join(basedir, basename.format(t=t, n=n))
+
+
+def file_path(basedir, basename, t, n=None):
```

Comment:
What's this needed for?  All we need to do is add an optional `restore=True|False` argument to `file_path_gen`.